src/main/java/by/home/repository/CircleRepositoryImpl.java

```diff
@@ -0,0 +1,58 @@
+package by.home.repository;
+
+import by.home.CircleObservable;
+import by.home.repository.specifications.Specification;
+import org.apache.log4j.Logger;
+
+import java.util.*;
+
+public class CircleRepositoryImpl implements CircleRepository {
+
+    private final static Logger LOGGER = Logger.getLogger(CircleRepositoryImpl.class);
+
+    private Map<Integer, CircleObservable> store;
+
+    public CircleRepositoryImpl(Map<Integer, CircleObservable> store) {
```

Comment:
make it empy, no need to pass map in constructor, use add to support encapsulation